app/com/linkedin/drelephant/spark/heuristics/UnifiedMemoryHeuristic.scala

```diff
@@ -74,6 +84,11 @@ object UnifiedMemoryHeuristic {
   val DEFAULT_SPARK_EXECUTOR_MEMORY_THRESHOLD = "2G"
   val UNIFIED_MEMORY_ALLOCATED_THRESHOLD = "256M"
   val SPARK_MEMORY_FRACTION_THRESHOLD : Double = 0.05
+  val JVM_USED_MEMORY = "jvmUsedMemory"
+  val MAX_EXECUTOR_PEAK_JVM_USED_MEMORY_THRESHOLD_KEY = "executor_peak_jvm_memory_threshold"
+
+  // 300 * FileUtils.ONE_MB (300 * 1024 * 1024)
```

Comment:
It looks like this is copying a lot of the code and logic from JvmUsedMemoryHeuristic.scala. Instead, could the code for evaluation be factored out and shared?